src/optimizer/projection_push_down_optimizer.cpp

```diff
@@ -44,18 +48,34 @@ void ProjectionPushDownOptimizer::visitOperator(LogicalOperator* op) {
 
 void ProjectionPushDownOptimizer::visitPathPropertyProbe(LogicalOperator* op) {
     auto& pathPropertyProbe = op->cast<LogicalPathPropertyProbe>();
-
-    if (pathPropertyProbe.getChild(0)->getOperatorType() == LogicalOperatorType::GDS_CALL) {
+    auto child = pathPropertyProbe.getChild(0);
+    if (child->getOperatorType() == LogicalOperatorType::GDS_CALL) {
         if (!nodeOrRelInUse.contains(pathPropertyProbe.getRel())) {
+            // Path is not needed
             pathPropertyProbe.setJoinType(planner::RecursiveJoinType::TRACK_NONE);
+            auto call = child->ptrCast<LogicalGDSCall>();
+            auto& info = call->getInfoUnsafe();
+            if (info.func.name == VarLenJoinsFunction::name) {
+                auto& rjAlg = info.func.gds->cast<RJAlgorithm>();
+                rjAlg.setToNoPath();
+                info.outExprs = rjAlg.getResultColumnsNoPath();
+            } else if (info.func.name == SingleSPPathsFunction::name) {
+                auto& rjAlg = info.func.gds->cast<RJAlgorithm>();
+                info.outExprs = rjAlg.getResultColumnsNoPath();
+                auto func = SingleSPDestinationsFunction::getFunction();
+                func.gds->setBindData(info.func.gds->getBindData()->copy());
+                info.func = std::move(func);
+            } else if (info.func.name == AllSPPathsFunction::name) {
+                auto& rjAlg = info.func.gds->cast<RJAlgorithm>();
+                info.outExprs = rjAlg.getResultColumnsNoPath();
+                auto func = AllSPDestinationsFunction::getFunction();
+                func.gds->setBindData(info.func.gds->getBindData()->copy());
+                info.func = std::move(func);
+            }
         }
         return;
     }
-
-    if (pathPropertyProbe.getChild(0)->getOperatorType() !=
-        planner::LogicalOperatorType::RECURSIVE_EXTEND) {
-        return;
-    }
+    KU_ASSERT(child->getOperatorType() == LogicalOperatorType::RECURSIVE_EXTEND);
```

Comment:
Will we keep the RECURSIVE_EXTEND operator after we deprecate?

Reply:
Yes